design-documents/graph-ql/coverage/shared-cart.md

```diff
@@ -1,30 +1,71 @@
+ Once we get this id it can be used on the rest of the cart mutations/queries. 
+ 
+ We have to add `cart_id` part of the Cart object, because we need to return the cart id in case of `customerCart()` and avoid a round trip call to `cart(cart_id)` if we were just to return a string from `customerCart()`
+ 
+ 
+ The proposal for this scenario reflected in schema is the following query:
 ```graphql
-customerCartId(): String!
+customerCart(): Cart! // where Cart! is the customer cart
 ```
-This operation will only work for registered customers (valid customer token must be provided in headers).
 
- ### Scenario
+### Scenario for point 1.
  
  In the following scenario a logged in user adds a product to the cart. After logging in to his account from the smartphone the cart still contains added product (actual GraphQL queries are replaced with pseudocode for simplicity): 
  
 | Step | Device     | Operation                                                                        | Headers        | Response                                          |
 |------|------------|----------------------------------------------------------------------------------|----------------|---------------------------------------------------|
-| 1    | Laptop     | customerCartId()                                                                 | customer-token | 123e4567              |
-| 2    | Laptop     | addProductsToCart("123e4567", [{"sku": "productA", "quantity": 2}]) {cart_id}    | customer-token | 123e4567              |
-| 3    | Smartphone | customerCartId()                                                                 | customer-token | 123e4567              |
-| 4    | Smartphone | getCart("123e4567") {cart_items {sku, quantity}}                                 | customer-token | {cart_items: [{"sku": "productA", "quantity": 2]} |
+| 1    | Laptop     | customerCart() { cart_id }                                                         | customer-token | { cart_id: "CustomerCart123e4567" }               |
+| 2    | Laptop     | addProductsToCart("CustomerCart123e4567", [{"sku": "productA", "quantity": 2}]) {cart_id, cart_items {sku, quantity} }    | customer-token | { cart_id: "CustomerCart123e4567", cart_items: [{"sku": "productA", "quantity": 2]}|
+| 3    | Smartphone | customerCart() { cart_id, cart_items {sku, quantity} }                                                         | customer-token | { cart_id: "CustomerCart123e4567", cart_items: [{"sku": "productA", "quantity": 2] }|
+| 4    | Smartphone | cart("CustomerCart123e4567") {cart_items {sku, quantity}}                        | customer-token | {cart_items: [{"sku": "productA", "quantity": 2]} |
+
+*Note 1*: customerCart will always return the same cart_id representing the same query, until an order is completed. If an active customer cart wasn't created, then one will be instantly created. So this could qualify as a mutation not a query.
```

Comment:
I believe it is better to convert this query to mutation because best practice is that queries do not modify the state.
We may also be able to combine it with "createEmptyCart"  in the future.